File: schedule/app/crawlSite/get_all_URLs.php
<?php


use Symfony\Component\DomCrawler\Crawler;

ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);



// Define the base URL and the list of relative URLs to scrape
$base_url = 'http://www.plan.pwsz.legnica.edu.pl/';
$urls = ['/',
    'schedule_view.php?site=show_kierunek.php&id=7',
    'schedule_view.php?site=show_kierunek.php&id=1',
    'schedule_view.php?site=show_kierunek.php&id=2',
    'schedule_view.php?site=show_kierunek.php&id=11',
    'schedule_view.php?site=show_kierunek.php&id=10',];

// Loop over the URLs and scrape data from each page
$file = fopen("/URLs.txt", "w");

foreach ($urls as $url) {
    $full_url = $base_url . $url;

    $query = parse_url($full_url, PHP_URL_QUERY);
    parse_str($query, $params);
    $id_faculty = $params['id'] ?? 0;
    fwrite($file, 'id_faculty='.$id_faculty . "\n");

    $grab = file_get_contents($full_url);
    $crawler = new Crawler($grab);

    // Traverse links from the Crawler
    foreach ($crawler->filter('a') as $link) {
        $data = $link->getAttribute('href');
        $full_link = $base_url . $data;
        file_put_contents('URLs.txt', $full_link);
        echo $full_link . "\n\n";
    }

}
fclose($file);

?>

File: schedule/app/crawlSite/get_all_URLs.py
import requests
from bs4 import BeautifulSoup
from urllib.parse import urljoin

# Define the base URL and the list of relative URLs to scrape
base_url = 'http://www.plan.pwsz.legnica.edu.pl/'
urls = ['schedule_view.php?site=show_kierunek.php&id=7']

# Loop over the URLs and scrape data from each page
with open("URLs_zdrikul.txt", "w") as f:
    for url in urls:
        full_url = urljoin(base_url, url)
        grab = requests.get(full_url)
        soup = BeautifulSoup(grab.text, 'html.parser')
        
        # traverse paragraphs from soup
        for link in soup.find_all("a"):
            data = link.get('href')
            full_link = urljoin(base_url, data)
            f.write(full_link)
            f.write("\n")
